lambda/src/server/main.cljs

```diff
@@ -28,23 +30,31 @@
   "Main AWS Lambda handler. Invoked by slackBot.
    See https://docs.aws.amazon.com/lambda/latest/dg/nodejs-handler.html"
   [^:js {:as req :keys [body]} ^js res]
-  (p/let [{:as result
-           :keys [response
-                  task]} (cond
-                           ;; Slack API: identification challenge
-                           (:challenge body) {:response (:challenge body)}
-
-                           ;; Slack Interaction (e.g. global shortcut)
-                           (:payload body) (handlers/handle-interaction! (:payload body))
-
-                           ;; Slack Event
-                           (:event body) (handlers/handle-event! (:event body)))]
-
-    (assert (or (map? result) (nil? result)))
-    (pp/pprint [(if result ::handled ::not-handled) body])
-    (when task (tasks/publish! task))
-
-    (.send res response)))
+  (if (:challenge body)
+    (.send res (:challenge body))
+    (p/let [[kind data team-id] (cond (:payload body) [:interaction (:payload body) (-> body :payload :team :id)]
+                                      (:event body) [:event (:event body) (:team_id body)]
+                                      (:challenge body) [:challenge (:challenge body) nil])
```

Comment:
We seem to handle the challenge in the outer `if` and then again here?

Reply:
ah right, i had tried both approaches and forgot to prune. pushed update.